### PR TITLE
feat: add phone-number package

### PR DESCRIPTION
**Pull request by alipourhabibi**
I added the phone_numbers package based on rust-persian-tools' data, along with tests for it.

### CODE REVIEW

File: phone_numbers/phone.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	}
+	return GetPrefixDetails(prefix)
+}
+
+// GetPrefixDetails returns the OperatorDetails of the given prefix
+func GetPrefixDetails(prefix string) (*OperatorDetails, error) {
+	for k, v := range MCIMap {
+		if k == prefix {
+			return &v, nil
+		}
```
Comment (nimahkh) [inline phone_numbers/phone.go:25]: This part is repeating, there is an opportunity to make it cleaner and DRY

**Review by nimahkh — APPROVED** [review]

File: phone_numbers/phone.go (hunk 1 above)
Comment (alipourhabibi) [reply phone_numbers/phone.go]: I used a map to store all operators and range over them to make it cleaner

Comment (nimahkh) [reply phone_numbers/phone.go]: Exactly what I meant, cool

**Review by nimahkh — APPROVED** [review]
LGTM 🚀

**Pull request merged**